tests/Feature/TransportTest.php

```diff
@@ -133,6 +135,24 @@ public function can_delay_messages_and_wait_for_receiving() {
         $this->then_the_queue_has_size(0);
     }
 
+    /** @test */
+    public function can_handle_debounce_stamp_messages() {
+        $this->given_there_is_a_wrapped_message();
+        $this->given_there_is_a_debounce_stamp_on_the_message('456', 100);
+        $this->when_the_message_is_sent_on_the_transport(5);
+        $this->then_the_queue_has_size(1);
+    }
+
+    /** @test */
+    public function can_debounce_messages_and_wait_for_receiving() {
+        $this->given_there_is_a_wrapped_message();
+        $this->given_there_is_a_debounce_stamp_on_the_message('7890', 100);
+        $this->when_the_message_is_sent_on_the_transport(1);
+        $this->then_another_message_is_sent_on_the_transport_with_same_stamp_after_a_delay_of(500);
```

Comment:
then's should only be used for assertions, this then is really more like a when. Your first when should exist as a given since it's a precondition for what we are testing.

Reply:
updated.